### PR TITLE
add UT for GetMinTolerationTime and GetMatchingTolerations

### PR DESCRIPTION
**Pull request by wongearl**
Signed-off-by: wongearl <vamliang@163.com>

**What type of PR is this?**

/kind cleanup
**What this PR does / why we need it**:
add UT for GetMinTolerationTime and GetMatchingTolerations
**Which issue(s) this PR fixes**:
Subtask of https://github.com/karmada-io/karmada/issues/2579
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/helper/taint_test.go [hunk 1]
```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+						TimeAdded: nil,
+					},
+				},
+				tolerations: nil,
+			},
+			want:  false,
+			want1: []corev1.Toleration{},
+		},
+		{
+			name: "test return tolerations",
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:517]: How about changing the test case name to `don't exist mismatch taint`?

And Adding a case `exist mismatch taint` case will be better.

File: pkg/util/helper/taint_test.go [hunk 2]
```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+		noExecuteTaints []corev1.Taint
+		usedTolerations []corev1.Toleration
+	}
+	tests := []struct {
+		name string
+		args args
+		want time.Duration
+	}{
+		{
+			name: "test no noExecute taints",
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:376]: How about changing the test case name to `the length of taints is 0`? Same as the other case names.

File: pkg/util/helper/taint_test.go [hunk 3]
```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:450]: Those taints and tolerations have the same key, value and operator, how about making them a little different?

File: pkg/util/helper/taint_test.go [hunk 4]
```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+				},
+			},
+			want: 9999999999999,
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:453]: I test the case failed on my local side:
```
taint_test.go:459: GetMinTolerationTime() = 10000000, want 9999999
```

I think it's best that we don't use the current time when we use the time: 

```go
# line 364
now := metav1.Now()
```

File: pkg/util/helper/taint_test.go [hunk 5]
```diff
@@ -357,3 +359,209 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
 }
+
+func TestGetMinTolerationTime(t *testing.T) {
+	now := metav1.Now()
+	later := metav1.NewTime(time.Now().Add(5000))
+	type args struct {
+		noExecuteTaints []corev1.Taint
+		usedTolerations []corev1.Toleration
+	}
+	tests := []struct {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:370]: I think it's necessary to add a use case to this test `exist mismatch taint`.

The current method may not take this into account in its implementation, and it assumes that all the taints are matched, which is actually the logic in the program, but as a tool method, this assumption is not reasonable, so the implementation of the method may need to be modified.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Hi @wongearl, thanks a lot~

**Comment by XiShanYongYe-Chang** [comment]
Kindly ping @wongearl

File: pkg/util/helper/taint_test.go (hunk 1 above)
Comment (wongearl) [reply pkg/util/helper/taint_test.go:517]: ok

File: pkg/util/helper/taint_test.go (hunk 2 above)
Comment (wongearl) [reply pkg/util/helper/taint_test.go:376]: ok

File: pkg/util/helper/taint_test.go (hunk 3 above)
Comment (wongearl) [reply pkg/util/helper/taint_test.go:450]: ok

File: pkg/util/helper/taint_test.go (hunk 4 above)
Comment (wongearl) [reply pkg/util/helper/taint_test.go:453]: ok

File: pkg/util/helper/taint_test.go [hunk 6]
```diff
@@ -357,3 +359,283 @@ func TestGetNoExecuteTaints(t *testing.T) {
...
+					{
+						Key:               "key5",
+						Operator:          "Equal",
+						Value:             "value5",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
+				},
+			},
+			want: -1,
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/taint_test.go:499]: This is not the expected value. The expected value should be 0. We need to modify the test method to meet this test case.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks for your quick response.

**Comment by wongearl** [comment]
sorry，I close it

**Pull request closed without merging**